src/mds/MDSRank.cc

```diff
@@ -1175,6 +1186,44 @@ bool MDSRank::_dispatch(const cref_t<Message> &m, bool new_msg)
   return true;
 }
 
+#ifdef WITH_CEPHFS_NOTIFICATION
+
+void MDSRank::send_notification_info_to_peers(const ref_t<Message> &m) {
+  set<mds_rank_t> up;
+  get_mds_map()->get_up_mds_set(up);
+  for (const auto &r : up) {
+    if (r == get_nodeid()) {
+      continue;
+    }
+    send_message_mds(m, r);
+  }
+}
+
+bool MDSRank::is_notification_info(const cref_t<Message> &m) {
```

Comment:
Perhaps it would be better to define this method unconditionally as well but handle most of its logic (add/remove and kafka/udp discrimination) inside notification_manager which is always created anyway?